### ISSUE TITLE
🐛 --repoter not working with `check` and `lint` command when lint from stdin

### ISSUE THREAD

**Issue by axetroy** [biomejs/biome#4045]
### Environment information

```block
echo "var a = 123" | npx @biomejs/biome lint --reporter=json--stdin-file-path=test.js
```

### What happened?

output to stdout

```
var a = 123
stdin ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  × The contents aren't fixed. Use the `--fix` flag to fix them.

```

### Expected result

`--repoter` should works

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Why should reporters work on stdin? That's by design.

**Comment by axetroy** [comment]
I don't know what the purpose of this design is.

But I think the output should be formatted by reporter.

For example, I got a scenario now. 

I want to lint a file. But since I have already read the file, I want to pass it in through stdin to save the another file reading.

**Comment by ematipico** [comment]
I didn't understand why a JSON reporter is needed in stdin. I understand you have a use case, but I honestly didn't understand what this use case does. 

Also, you expect the `--reporter` to work, but you didn't explain what you expect to get from the output. At the moment, when you call a command without `--write`, the stdin mode return the original code.

**Comment by Conaclos** [comment]
Is there a particular reason to not make `--reporter` compatible with stdin + `--stdin-file-path`?

**Comment by ematipico** [comment]
If we decide to do so, we need to explain to users why we implemented and what's a valid use case. I'm not against implementing stuff, but we need to be good at justifying things so we don't regret them in the future. That's part of our philosophy

**Comment by axetroy** [comment]
> we need to explain to users why we implemented and what's a valid use case.

For keeping `--reporter` consistent. When this flag is specified, all output should be formatted by the reporter, without any exceptions.
